src/list.tsx

```diff
@@ -242,7 +218,30 @@ const Content: React.FC<ContentProps> = ({
                 </Stack>
             </div>
             {renderBadge()}
-            {renderRight()}
+            {type === 'chevron' && (
+                <Box paddingLeft={16} className={classNames(classes.center, {[classes.disabled]: disabled})}>
+                    <IconChevron
+                        color={isInverse ? colors.inverse : colors.neutralMedium}
+                        direction="right"
+                    />
+                </Box>
+            )}
+            {type === 'control' && <div className={classes.right}>{renderRight?.({centerY})}</div>}
+            {type === 'custom' && (
+                <div className={classNames(classes.right, {[classes.disabled]: disabled})}>
+                    {renderRight?.({centerY})}
+                </div>
+            )}
+            {type === 'custom' && navigable === true && (
+                <div className={classNames(classes.right, {[classes.disabled]: disabled})}>
+                    <div style={centerY ? {display: 'flex', alignItems: 'center', height: '100%'} : {}}>
+                        <IconChevron
+                            color={isInverse ? colors.inverse : colors.neutralMedium}
+                            direction="right"
+                        />
+                    </div>
+                </div>
+            )}
```

Comment:
This should be equal, while narrowing the conditional renderings:

```suggestion
            {type === 'custom' && (
                <>
                    <div className={classNames(classes.right, {[classes.disabled]: disabled})}>
                        {renderRight?.({centerY})}
                    </div>
                    {navigable && (
                        <div className={classNames(classes.right, {[classes.disabled]: disabled})}>
                            <div
                                style={centerY ? {display: 'flex', alignItems: 'center', height: '100%'} : {}}
                            >
                                <IconChevron
                                    color={isInverse ? colors.inverse : colors.neutralMedium}
                                    direction="right"
                                />
                            </div>
                        </div>
                    )}
                </>
            )}
```